src_features/generic_tx_parser/gtp_field_table.c

```diff
@@ -0,0 +1,99 @@
+#ifdef HAVE_GENERIC_TX_PARSER
+
+#include <string.h>
+#include "os_print.h"
+#include "gtp_field_table.h"
+#include "mem.h"
+
+// type (1) | key_length (1) | value_length (2) | key ($key_length) | value ($value_length)
+
+typedef struct {
+    const uint8_t *start;
+    size_t size;
+} s_field_table;
+
+static s_field_table g_table;
+
+void field_table_init(void) {
+    explicit_bzero(&g_table, sizeof(g_table));
+}
+
+const uint8_t *get_next_table_entry(const uint8_t *ptr, s_field_table_entry *entry) {
+    uint8_t key_len;
+    uint16_t value_len;
+
+    if (entry != NULL) {
+        memcpy(&entry->type, ptr, sizeof(entry->type));
+    }
+    ptr += sizeof(entry->type);
+    memcpy(&key_len, ptr, sizeof(key_len));
+    ptr += sizeof(key_len);
+    memcpy(&value_len, ptr, sizeof(value_len));
+    ptr += sizeof(value_len);
+    if (entry != NULL) {
+        entry->key = (char *) ptr;
+    }
+    ptr += key_len;
+    if (entry != NULL) {
+        entry->value = (char *) ptr;
+    }
+    ptr += value_len;
+    return ptr;
+}
```

Comment:
Not sure to understand the behavior when `entry == NULL` there. Do you want to be able to get a pointer to the next entry without copying the current value? However, if `entry` is indeed NULL, the line `ptr += sizeof(entry->type);` will still be executed resulting in a segmentation fault. Since we're leveraging the memory allocator, could we just store the next object pointer in the previous struct like a linked list? Seems simpler and better in term of performance.

Reply:
Oh! This is true, my bad!